src/mscorlib/src/System/Char.cs

```diff
@@ -927,12 +927,17 @@ public static String ConvertFromUtf32(int utf32)
                 // This is a BMP character.
                 return (Char.ToString((char)utf32));
             }
-            // This is a sumplementary character.  Convert it to a surrogate pair in UTF-16.
-            utf32 -= UNICODE_PLANE01_START;
-            char[] surrogate = new char[2];
-            surrogate[0] = (char)((utf32 / 0x400) + (int)CharUnicodeInfo.HIGH_SURROGATE_START);
-            surrogate[1] = (char)((utf32 % 0x400) + (int)CharUnicodeInfo.LOW_SURROGATE_START);
-            return (new String(surrogate));
+
+            unsafe
+            {
+                // This is a supplementary character.  Convert it to a surrogate pair in UTF-16.
+                utf32 -= UNICODE_PLANE01_START;
+                uint surrogate = 0; // allocate 2 chars worth of stack space
+                char* address = (char*)&surrogate; // we forego using stackalloc here because it's slow; see dotnet/coreclr#5719
```

Comment:
I do not think that the comment about stackalloc is relevant here. It would be generally better to allocate the space as `uint` here regardless - given how the optimizations that the JIT is capable of.

A different problem is that dotnet/coreclr#5719 is pretty confused. I do not think it is a real problem. I have just added a comment to it. We have been using stackalloc for years for unsafe perf optimizations with pretty clear results.